File: AgoraChat/Code/Controllers/Settings/ACDPrivacyViewController.m
//
//  EMSecurityPrivacyViewController.m
//  EaseIM
//
//  Created by 娜塔莎 on 2020/6/10.
//  Copyright © 2020 娜塔莎. All rights reserved.
//

#import "ACDPrivacyViewController.h"
#import "ACDBlockListViewController.h"
#import "ACDTitleDetailCell.h"

@interface ACDPrivacyViewController ()

@end

@implementation ACDPrivacyViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.navigationItem.leftBarButtonItem = [ACDUtil customLeftButtonItem:@"Privacy" action:@selector(back) actionTarget:self];

    [self _setupSubviews];
}

- (void)_setupSubviews
{
    self.view.backgroundColor = [UIColor colorWithRed:249/255.0 green:249/255.0 blue:249/255.0 alpha:1.0];

    self.table.scrollEnabled = NO;
    self.table.rowHeight = 66;
    self.table.tableFooterView = [[UIView alloc] init];
    self.table.backgroundColor = [UIColor whiteColor];
    self.table.scrollEnabled = NO;
    self.table.separatorStyle = UITableViewCellSeparatorStyleNone;
    [self.table mas_makeConstraints:^(MASConstraintMaker *make) {
        make.top.equalTo(self.view);
        make.left.equalTo(self.view);
        make.right.equalTo(self.view);
        make.height.equalTo(@70);
    }];

}

- (void)back {
    [self.navigationController popViewControllerAnimated:YES];
}


#pragma mark - Table view data source
- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return 1;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {

    ACDTitleDetailCell *cell = [tableView dequeueReusableCellWithIdentifier:[ACDTitleDetailCell reuseIdentifier]];
    if (cell == nil) {
        cell = [[ACDTitleDetailCell alloc] initWithStyle:UITableViewCellStyleValue1 reuseIdentifier:[ACDTitleDetailCell reuseIdentifier]];
        cell.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
    }

    if (indexPath.row == 0) {
        cell.nameLabel.text = NSLocalizedString(@"Blocked List", nil);
        ACD_WS
        cell.tapCellBlock = ^{
            [weakSelf goBlockListPage];
        };
    }
    return cell;
}

- (void)goBlockListPage {
    ACDBlockListViewController *controller = [[ACDBlockListViewController alloc] init];
    [self.navigationController pushViewController:controller animated:YES];

}

#pragma mark - Table view delegate

- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath
{
    return 54.0;
}


- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
    NSInteger section = indexPath.section;
    if (section == 0) {
    }
}

@end

File: AgoraChat-Swift/AgoraChat-Swift/code/Controllers/Settings/ACDPrivacyViewController.swift
//
//  ACDPrivacyViewController.swift
//  AgoraChat-Swift
//
//  Created by 冯钊 on 2022/10/13.
//

import UIKit
import SnapKit

class ACDPrivacyViewController: ACDBaseTableViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        self.navigationItem.leftBarButtonItem = ACDUtil.customLeftButtonItem(title: "Privacy".localized, action: #selector(back), target: self)
        
        self.view.backgroundColor = Color_F9F9F9

        self.table.isScrollEnabled = false
        self.table.rowHeight = 54
        self.table.tableFooterView = UIView()
        self.table.backgroundColor = .white
        self.table.separatorStyle = .none
        self.table.snp.remakeConstraints { make in
            make.top.left.right.bottom.equalTo(self.view)
        }
        self.table.register(UINib(nibName: "ACDSubDetailCell", bundle: nil), forCellReuseIdentifier: "cell")
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        self.navigationController?.setNavigationBarHidden(false, animated: true)
    }
    
    private func goBlockListPage() {
        let vc = ACDBlockListViewController()
        self.navigationController?.pushViewController(vc, animated: true)
    }
    
    @objc private func back() {
        self.navigationController?.popViewController(animated: true)
    }
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return 1
    }
 
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "cell", for: indexPath)
        cell.accessoryType = .disclosureIndicator
        if let cell = cell as? ACDSubDetailCell {
            cell.showSubDetailLabel = false
            cell.nameLabel.text = "Blocked List".localized
            cell.tapCellHandle = { [unowned self] in
                self.goBlockListPage()
            }
        }
        
        return cell
    }
}
